### DISCUSSION TITLE
Watch Pending Transactions

### DISCUSSION THREAD

**Discussion by real-lazy-coder** [wevm/viem#1553] [Question] [answered]
Reading through the code it looks like the websocket client covers the newPendingTransactions subscription. I'm looking to implement the alchemy_pendingTransactions subscription. This is a subscription unique to alchemy. 

Before I go through the process of creating a PR to add the feature, is there another avenue I should look into to achieve this?

Is there a way to subscribe to a custom message via a viem client?

**Comment by jxom** [comment, marked as answer]
You can use the raw subscribe method from the websocket transport... if you're using a websocket transport, you can do 

```ts
client.transport.subscribe({
  params: ['alchemy_pendingTransactions'],
  onData(data) {
    console.log(data)
  }
})
```

**Reply by real-lazy-coder** [reply]
Much appreciated. I didn't realize this was accessible. 🙏
